modin/pandas/test/test_io.py

```diff
@@ -199,6 +200,26 @@ def eval_to_file(modin_obj, pandas_obj, fn, extension, **fn_kwargs):
         teardown_test_files([unique_filename_modin, unique_filename_pandas])
 
 
+@pytest.fixture
+def make_parquet_dir():
+    list_of_path: list[str] = []
+
+    def _make_parquet_dir(
+        path: str, dfs_by_filename: Dict[str, pandas.DataFrame], row_group_size: int
+    ):
+        if os.path.exists(path):
+            shutil.rmtree(path)
+        else:
+            os.makedirs(path)
+        for filename, df in dfs_by_filename.items():
+            df.to_parquet(os.path.join(path, filename), row_group_size=row_group_size)
+        list_of_path.append(path)
+
+    yield _make_parquet_dir
+
+    shutil.rmtree(list_of_path[0])
```

Comment:
This appending (to the end) and removal of the first (but not removing this first item from the list) all looks really weird.
What was wrong with just `shutil.rmtree(path)` _without_ keeping a `list_of_path` variable around?..

Reply:
I copied this pattern from the existing [`make_parquet_file`](https://github.com/modin-project/modin/blob/71815691339ea5a61be4c2f3e448fd51ea97b77e/modin/conftest.py#L437). The inner function that we `yield` has to communicate the `path` up to the `make_parquet_dir` for cleanup. The simplest way I could think of at the time was to mutate a list that's in the `make_parquet_dir` scope.

Reply:
I'm not entirely sure.  But looking at this now I think we can just use something like `tempdir` for all of this logic.